### PR TITLE
Virtual Card: Options, Policy and Request a Card

### PR DESCRIPTION
**Pull request by kewitz**

### CODE REVIEW

File: server/graphql/common/expenses.ts [hunk 1]
```diff
@@ -676,14 +677,22 @@ export async function editExpense(
...
     throw new Unauthorized("You don't have permission to edit this expense");
   }
 
-  const isPaidCreditCardCharge = expense.type === expenseType.CHARGE && expense.status === expenseStatus.PAID;
+  const isPaidCreditCardCharge =
+    expense.type === expenseType.CHARGE && expense.status === expenseStatus.PAID && Boolean(expense.VirtualCardId);
+
+  if (isPaidCreditCardCharge && !hasItemChanges) {
+    throw new ValidationFailed(
+      'You need to include Expense Items when adding missing information to card charge expenes',
```
Comment (SudharakaP) [inline server/graphql/common/expenses.ts:693]:
```suggestion
      'You need to include Expense Items when adding missing information to card charge expenses',
```

File: server/graphql/common/expenses.ts [hunk 2]
```diff
@@ -769,21 +778,30 @@ export async function editExpense(
...
+      status: shouldUpdateStatus ? 'PENDING' : expense.status,
+      FromCollectiveId: fromCollective.id,
+      PayoutMethodId: PayoutMethodId,
+      legacyPayoutMethod: models.Expense.getLegacyPayoutMethodTypeFromPayoutMethod(payoutMethod),
+      tags: cleanExpenseData.tags,
+    };
+    if (isPaidCreditCardCharge) {
+      updatedExpenseProps['data'] = { ...expense.data, missingDetails: false };
+    }
+    return expense.update(updatedExpenseProps, { transaction: t });
```
Comment (SudharakaP) [inline server/graphql/common/expenses.ts:794]: Maybe rename, `t` to transaction and just put, 🤔 

```suggestion
    return expense.update(updatedExpenseProps, { transaction });
```

Comment (kewitz) [reply server/graphql/common/expenses.ts:794]: The problem here is that "Transaction" may suggest something very different given the context.

Comment (SudharakaP) [reply server/graphql/common/expenses.ts:794]: Ah, I see; then we could leave it I guess. Just that looks odd to have a one letter argument. 😸 😸

**Pull request merged**